gopher/parsers/tabular.py

```diff
@@ -66,25 +67,46 @@ def read_diann(proteins_tsv: str) -> pd.DataFrame:
 
     Args:
         proteins_tsv (str): Path to the DIANN-generated TSV file.
+            Expected columns:
+                'Protein.Group',
+                'Protein.Ids',
+                'Protein.Names',
+                'Genes',
+                'First.Protein.Description',
+                <several MSR columns>
+
 
     Returns:
         pd.DataFrame: A DataFrame with the processed protein data, indexed by
             the first protein accession.
-            The returned DataFrame excludes the following columns:
-            ["Protein.Group", "Protein.Ids", "Protein.Names", "Genes",
-            "First.Protein.Description"].
+            The returned DataFrame has the "Protein.Ids" column as the 
+            index and all columns are the MSR columns.          
     """
     proteins = pd.read_table(proteins_tsv)
     accessions = proteins["Protein.Ids"].str.split(";").str[0]
 
     proteins = proteins.set_index(accessions)
     proteins = proteins.rename_axis("Protein", axis="index")
-    return proteins.drop(
+    proteins = proteins.drop(
         columns=[
             "Protein.Group",
             "Protein.Ids",
             "Protein.Names",
             "Genes",
             "First.Protein.Description",
         ]
-    )
\ No newline at end of file
+    )
+
+    # Check data types
+    # (if loading from S3, default types are 'O'
+    if proteins.index.dtype not in ["O", "category", "str"]:
```

Comment:
why did you go with this instead of trying to cast to a number and fail if it cannot?